### PR TITLE
Box component refactor

### PR DESCRIPTION
**Pull request by Malinicz**
- make Box component deprecated
- create new Box component, which is simple and provides only 4 things: padding, background color, border and shadow. 

image
image
image

### CODE REVIEW

File: src/components/box/BoxDeprecated.jsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
+// @flow strict
```
Comment (Malinicz) [inline src/components/box/BoxDeprecated.jsx:1]: this is old Box without changes

File: src/components/box/_box-deprecated.scss [hunk 2]
```diff
@@ -0,0 +1,165 @@
+$boxBorderColor: $graySecondary;
```
Comment (Malinicz) [inline src/components/box/_box-deprecated.scss:1]: old styles, without change

File: src/components/box/pages/box-deprecated.jsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
+import React from 'react';
```
Comment (Malinicz) [inline src/components/box/pages/box-deprecated.jsx:1]: no change

File: src/components/box/pages/box-deprecated-interactive.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
```
Comment (Malinicz) [inline src/components/box/pages/box-deprecated-interactive.jsx:1]: no change

File: src/components/box/Box.spec.jsx [hunk 5]
```diff
@@ -29,128 +28,82 @@ test('border', () => {
   const box = shallow(boxComponent);
 
   expect(boxComponent.props.border).toEqual(true);
-  expect(box.hasClass('sg-box--no-border')).toEqual(false);
+  expect(box.hasClass('sg-box--with-border')).toEqual(true);
```
Comment (matzimowski) [inline src/components/box/Box.spec.jsx:31]: maybe we could get rid of `with`, wdyt? just `--border`

Comment (Malinicz) [reply src/components/box/Box.spec.jsx]: @matzimowski https://github.com/brainly/style-guide/pull/1864/commits/5b81d331777a0660cb07981de44408846a89ee50

File: src/components/box/Box.jsx [hunk 6]
```diff
@@ -23,246 +22,129 @@ type ColorType =
...
-   *          // child from box below won't be rendered
-   *          <Box imgSrc="https://some_url">
-   *            THIS CONTENT WILL
-   *            BE REPLACED BY IMAGE
-   *          </Box>
+   * Children to be rendered inside of the Box
+   * @example <Box>Text inside Box</Box>
    */
-  children?: ?Node,
+  children: ?Node,
```
Comment (bartoszlorek) [inline src/components/box/Box.jsx:69]: just `children: Node` because it's a enum that contains null

File: src/components/box/Box.jsx [hunk 7]
```diff
@@ -23,246 +22,129 @@ type ColorType =
...
-   * @see color=blue-secondary https://styleguide.brainly.com/latest/docs/interactive.html?color=blue-secondary#boxes
-   * @see color=blue-secondary-light https://styleguide.brainly.com/latest/docs/interactive.html?color=blue-secondary-light#boxes
-   * @see color=gray-secondary-lightest https://styleguide.brainly.com/latest/docs/interactive.html?color=gray-secondary-lightest#boxes
-   * @see color=gray-secondary-ultra-light https://styleguide.brainly.com/latest/docs/interactive.html?color=gray-secondary-ultra-light#boxes
-   * @see color=peach https://styleguide.brainly.com/latest/docs/interactive.html?color=peach#boxes
-   * @see color=mustard-primary https://styleguide.brainly.com/latest/docs/interactive.html?color=mustard-primary#boxes
+   * Box background color
+   * @example <Box color="mint-secondary">Text on a mint background</Box>
    */
   color?: ?ColorType,
```
Comment (bartoszlorek) [inline src/components/box/Box.jsx:80]: `color?: ColorType` without maybe

File: src/components/box/Box.jsx [hunk 8]
```diff
@@ -23,246 +22,129 @@ type ColorType =
...
-  padding?: ?PaddingType,
-  /**
-   * Specify src for image, which will be displayed INSTEAD of inner content
-   * @example <Box imgSrc="https://some_url">
-   *            THIS CONTENT WILL
-   *            BE REPLACED BY IMAGE
-   *          </Box>
-   */
-  imgSrc?: ?string,
+  shadow?: ?boolean,
```
Comment (bartoszlorek) [inline src/components/box/Box.jsx:87]: `shadow?: boolean` without maybe

File: src/components/box/Box.jsx [hunk 9]
```diff
@@ -23,246 +22,129 @@ type ColorType =
...
-   * @example <Box noBorderRadius>
-   *            some text
-   *          </Box>
-   * @see noBorderRadius=true https://styleguide.brainly.com/latest/docs/interactive.html?noBorderRadius=true#boxes
-   * @see noBorderRadius=false https://styleguide.brainly.com/latest/docs/interactive.html?noBorderRadius=false#boxes
+   * Disable border radius
+   * @example <Box noBorderRadius>Text inside Box with no border radius</Box>
    * @default false
    */
   noBorderRadius?: ?boolean,
```
Comment (bartoszlorek) [inline src/components/box/Box.jsx:100]: `noBorderRadius?: boolean` without maybe

**Pull request merged**